crates/uv-workspace/src/settings.rs

```diff
@@ -0,0 +1,84 @@
+use serde::Deserialize;
+use std::path::PathBuf;
+
+use distribution_types::{FlatIndexLocation, IndexUrl};
+use install_wheel_rs::linker::LinkMode;
+use uv_auth::KeyringProvider;
+use uv_configuration::{ConfigSettings, IndexStrategy, PackageNameSpecifier};
+use uv_normalize::PackageName;
+use uv_resolver::{AnnotationStyle, ExcludeNewer, PreReleaseMode, ResolutionMode};
+use uv_toolchain::PythonVersion;
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, Default, Deserialize)]
+pub(crate) struct PyProjectToml {
+    pub(crate) tool: Option<Tools>,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, Default, Deserialize)]
+pub(crate) struct Tools {
+    pub(crate) uv: Option<Options>,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, Default, Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub struct Options {
+    pub quiet: Option<bool>,
+    pub verbose: Option<bool>,
+    pub native_tls: Option<bool>,
+    pub no_cache: bool,
+    pub cache_dir: Option<PathBuf>,
+    pub pip: Option<PipOptions>,
+    pub resolver: Option<ResolverOptions>,
+    pub installer: Option<InstallerOptions>,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, Default, Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub struct PipOptions {
+    pub system: Option<bool>,
+    pub offline: Option<bool>,
+    pub index_url: Option<IndexUrl>,
+    pub extra_index_url: Option<IndexUrl>,
+    pub no_index: Option<bool>,
+    pub find_links: Option<Vec<FlatIndexLocation>>,
+    pub index_strategy: Option<IndexStrategy>,
+    pub keyring_provider: Option<KeyringProvider>,
+    pub no_build: Option<bool>,
+    pub no_binary: Option<Vec<PackageNameSpecifier>>,
+    pub only_binary: Option<Vec<PackageNameSpecifier>>,
+    pub no_build_isolation: Option<bool>,
+    pub resolver: Option<ResolverOptions>,
+    pub installer: Option<InstallerOptions>,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, Default, Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub struct ResolverOptions {
+    pub resolution: Option<ResolutionMode>,
+    pub prerelease: Option<PreReleaseMode>,
+    pub no_strip_extras: Option<bool>,
+    pub no_annotate: Option<bool>,
+    pub no_header: Option<bool>,
+    pub generate_hashes: Option<bool>,
+    pub legacy_setup_py: Option<bool>,
+    pub config_setting: Option<ConfigSettings>,
+    pub python_version: Option<PythonVersion>,
+    pub exclude_newer: Option<ExcludeNewer>,
+    pub no_emit_package: Option<Vec<PackageName>>,
+    pub emit_index_url: Option<bool>,
+    pub emit_find_links: Option<bool>,
+    pub annotation_style: Option<AnnotationStyle>,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, Default, Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub struct InstallerOptions {
+    pub link_mode: Option<LinkMode>,
+    pub compile: Option<bool>,
```

Comment:
I wonder if we should choose a clearer name for this

Reply:
`compile_bytecode`? We could change the CLI flag too (and add an alias).

Reply:
That makes sense to me.